paddle/fluid/recordio/scanner.h

```diff
@@ -0,0 +1,44 @@
+//   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <fstream>
+#include <memory>
+#include "paddle/fluid/recordio/chunk.h"
+namespace paddle {
+namespace recordio {
+
+class Scanner {
```

Comment:
In previous implementation, `Scanner`is a concept refer to  `multi-files reader`, it holds multiple files, and each `range_scanner` construct the index and parse a single recordio file. At the same time, benefits from the range_scanner, a scanner can scan through a list of files and generates a sequence of records.
In this PR, `Scanner` is a `Reader` or `Parser` to a single recordio file.
Should we change it to another name or leave a comment? `Scanner` as a single file reader seems confused.